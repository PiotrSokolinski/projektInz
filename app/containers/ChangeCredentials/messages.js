/*
 * ChangeCredentials Messages
 *
 * This contains all the text for the ChangeCredentials container.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.containers.ChangeCredentials'

export default defineMessages({
  confrimPasswordLabel: {
    id: `${scope}.confrimPasswordLabel`,
    defaultMessage: 'Your password ',
  },
  authorizeUser: {
    id: `${scope}.authorizeUser`,
    defaultMessage: 'Authorize user',
  },
})
