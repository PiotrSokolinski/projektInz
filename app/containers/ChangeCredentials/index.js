/**
 *
 * ChangeCredentials
 *
 */

import React, { useState } from 'react'
import { FormattedMessage, injectIntl } from 'react-intl'
import PropTypes from 'prop-types'

import Input from 'components/Input'
import Button from 'components/Button'
import ChangePassword from 'containers/ChangePassword'
import ChangeMail from 'containers/ChangeMail'

import messages from './messages'
import * as Styled from './styled'

const ChangeCredentials = ({ intl, type, onClose }) => {
  const [isResetting, setIsResetting] = useState(false)
  const [password, setPassword] = useState('')
  const confirmPassword = () => {
    if (password.length >= 8) setIsResetting(true)
  }
  return (
    <Styled.Container>
      {!isResetting && (
        <React.Fragment>
          <Input
            id="confirmPasswordField"
            label={intl.formatMessage(messages.confrimPasswordLabel)}
            inputProps={{
              type: 'password',
              name: 'confirmPassword',
              value: password,
              onChange: e => setPassword(e.target.value),
            }}
          />
          <Button type="submit" onClick={confirmPassword}>
            <FormattedMessage {...messages.authorizeUser} />
          </Button>
        </React.Fragment>
      )}
      {isResetting && (
        <React.Fragment>
          {type === 'email' ? (
            <ChangeMail password={password} onClose={onClose} />
          ) : (
            <ChangePassword password={password} onClose={onClose} />
          )}
        </React.Fragment>
      )}
    </Styled.Container>
  )
}

ChangeCredentials.propTypes = {
  intl: PropTypes.object,
  children: PropTypes.node,
}

export default injectIntl(ChangeCredentials)
