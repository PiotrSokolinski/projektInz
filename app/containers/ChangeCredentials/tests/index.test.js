/**
 *
 * Tests for ChangeCredentials
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'

import ChangeCredentials from '../index'
import { DEFAULT_LOCALE } from '../../../i18n'

describe('<ChangeCredentials />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <ChangeCredentials />
      </IntlProvider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <ChangeCredentials />
      </IntlProvider>,
    )
    expect(firstChild).toMatchSnapshot()
  })
})
