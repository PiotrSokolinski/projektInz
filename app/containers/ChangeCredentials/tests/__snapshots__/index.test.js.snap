// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangeCredentials /> Should render and match the snapshot 1`] = `
<div
  class="styled__Container-makxqa-0 jbfwFg"
>
  <div
    class="field"
    id="confirmPasswordField"
  >
    <label
      class="styled__StyledLabel-oaisvy-0 fCsvBu"
    >
      Your password 
    </label>
    <div
      class="control"
    >
      <input
        class="styled__Input-ms4xi4-0 jirwCh"
        name="confirmPassword"
        type="password"
        value=""
      />
    </div>
    
  </div>
  <button
    class="styled__StyledButton-sc-1gb27vo-0 cSvxQx"
    type="submit"
  >
    <span>
      Authorize user
    </span>
  </button>
</div>
`;
